packages/frontend/src/components/common/Filters/configs.ts

```diff
@@ -122,8 +124,8 @@ type FilterRuleLabels = {
     value?: string;
 };
 
-export const getFilterRuleLabel = (
-    filterRule: FilterRule,
+export const getConditionalRuleLabel = (
```

Comment:
do we have tests for these labels? I'm finding it hard to visualise how the final labels will look in each use case!

Reply:
@owlas no we don't. do you think it is a good idea to add tests in this PR?

I was thinking to split this function into two actually. first for the operator (textual representation of the operator) and the second for the value and then join. will be much more easier to read and understand. right now it looks very chaotic

Reply:
I did the splitting part but test is out of the scope of the PR